### ISSUE TITLE
Add commands to refresh addons & gamemodes to servers for new files

### ISSUE THREAD

**Issue by Kitty1789** [Facepunch/garrysmod-requests#832]
Recently there was a client command added "reload_legacy_addons" or so, and you could add it to the server, so it reloads the /addons/ and the /gamemodes/ folders, and "adds + sends" the files to the client & server, as if the server would reboot, it would remove the need of a full reboot, and allow owners to instantly add new addons (if the "new" addons adds models/textures, simply use the error texture until the player rejoins to apply the model, if you cant make him download it in-game), it would make developing easier, sometimes you maybe want to add another file for a small gamemode code (for organization, as seen in most gamemodes) wich you'd like to implement ingame without the need of a full reboot.

Hope you get what I mean!

**Comment by LuaTenshi** [comment]
Changing level is your best bet currently. Also we do have... https://wiki.garrysmod.com/page/Auto_Refresh

**Comment by WhiteKitty00** [comment]
Yeah I know about auto refresh already etc, but level changing/autorefresh wont check for NEWLY MADE files, You've to make a full server reboot for new files to be applied.

**Comment by fruitwasp** [comment]
1. restart your server daily during the night. 
2. execute RunString in the server console to apply emergency patches.

**Comment by Acecool** [comment]
Sorry for the wall of text; in short... I Agree... This should be a default dev concommand to reload your gm files ( but it is super easy to add yourself )...

I've asked for an intelligent auto-refresh for a while but it wasn't added so I added the feature to my autoloader... It is many times faster to load the altered file ( by comparing ALL files vs filetime ) that it is to reload all files in a gm.. It networks the data to users currently in game ( I haven't tested to see if new users download the newest file or not because I don't have a mac to test with but hopefully someone can help me out to see if the strings table updates )

Unfortunately, there are still issues with autorefresh that makes me not want to use it and to only use my autoloader... basically there are times when the base gamemode is reloaded despite no files changing ( and I only call DeriveGamemode once per level ) and autorefresh being called multiple times over 1 file being saved ( which led to me incorporating a don't-load-clause if already mid auto-refresh but because the base game-mode gets reloaded even though no changes have been made, it overwrites the functions and you end up having to reload the loaded game-mode... this caused other issues and I removed it for the time being )...

For my latest autoloader; probably the 10th iteration which I am hoping to release soon with my framework ( which can be used with it, or standalone because of the instruction system I've added ) separates out all of the game-mode data and rewrites GM into a meta table to call functions out of 3 distinct tables which lets me prevent data from being overwritten when not necessary among other things ( such as loading / unloading game-modes in real-time )... basically: there are 3 tables.. GM_BASE, GM_FRAMEWORK, and GM_LOADED... base for base / sandbox, framework for the framework changes ( which I am reducing to only necessary changes to correct behavior or smooth things out ), and loaded which is the gamemode which loads on top of the framework... By doing it this way it doesn't matter if the base loads last or first, etc.. they're loaded into proper tables and the meta table calls them in reverse order ( loaded, framework, or base ) depending on whether or not the function / variable / etc.. exists...

I honestly think GMod could benefit from integrating such a system by default ( The smart refresh and the 3 table design reworked into a 2 table design for base and current gm; frameworks can easily extend that )...
